types/ramda/index.d.ts

```diff
@@ -87,11 +87,11 @@ export function add(a: number): (b: number) => number;
  * Creates a new list iteration function from an existing one by adding two new parameters to its callback
  * function: the current index, and the entire list.
  */
-export function addIndex<T, U>(fn: (f: (item: T) => U, list: readonly T[]) => U[]): _.F.Curry<(a: (item: T, idx: number, list?: T[]) => U, b: readonly T[]) => U[]>;
+export function addIndex<T, U>(fn: (f: (item: T) => U, list: readonly T[]) => U[]): _.F.Curry<(a: (item: T, idx: number, list: T[]) => U, b: readonly T[]) => U[]>;
```

Comment:
In most time, people only want to add `index` when iterate something.

Reply:
is that related to this change though? or...

Reply:
or, well... is `undefined` ever passed as the `list` argument? otherwise it's safe to pass functions with fewer parameters: [playground](https://www.typescriptlang.org/play?jsx=0#code/MYewdgzgLgBAZgLhgHgCoD4AUBLKBTAWyVQBoZsATADyTAFcCAjPAJzIBttpiBtAXQCUMALzoYANxCURMHPgJlKVIaJgBvGAHoAVDAogYEEATxQAFtjABzGNs0wAvkA)

Reply:
Because curried function will only be invoked when you pass enough parameters, `curry` will calculate the fn's length by `Function.prototype.length`, which will ignore optional and rest parameters: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Length#using_function_length

Reply:
but this isn't `F.Curry`-ied, right? since it's nested? idk... if you can give a testcase where it fails that'd be nice

(i tested with passing `R.curry(fn)` as `a` and it seems to still typecheck fine)

Reply:
Oh, I see, it's a callback, maybe ts not support well enough to strict parameters' count.
I still think it's not nessary to remove list's `?`.

Reply:
oops, didn't see this comment...
i had to remove the `?` otherwise 3-argument callbacks get `T[] | undefined` as third argument, which is the wrong type.
`_.F.Curry` seems to handle `?:` fine, but as said above i believe the callback isn't `_.F.Curry`-ied
(however it doesn't hurt to do more testing to make sure nothing breaks)

Reply:
OK